__tests__/ExpensiMark-HTML-test.js

```diff
@@ -1934,4 +1931,10 @@ describe('Image markdown conversion to html tag', () => {
         const resultString = '<img src=\"https://example.com/image.png\" alt=\"test&quot; onerror=&quot;alert(&#x27;xss&#x27;)\" />';
         expect(parser.replace(testString)).toBe(resultString);
     });
+
+    test('No html inside the src attribute', () => {
+        const testString = '![`code`](https://example.com/image.png)';
+        const resultString = '<img src="https://example.com/image.png" alt="<code>code</code>" />';
+        expect(parser.replace(testString)).toBe(resultString);
+    })
```

Comment:
@kidroca Shouldn't it be `alt="&lt;code&gt;code&lt;/code&gt;"` instead of `alt="<code>code</code>"`?

Reply:
> How does this impact Live Markdown functionality?

Currently, Live Markdown uses pretty basic algorithm for parsing HTML that alternates between finding `<` and then `>`.

The output `<img src="..." alt="<code>...</code>" ... />` breaks the parser as the `>` from `<code>` gets matched instead of the (self-)closing `>` from `img` tag.

According to the [HTML standard](https://html.spec.whatwg.org/multipage/syntax.html#attributes-2), `>` is not a valid character in HTML attribute:

> Attributes have a name and a value. Attribute names must consist of one or more characters other than [controls](https://infra.spec.whatwg.org/#control), U+0020 SPACE, U+0022 ("), U+0027 ('), U+003E (>), U+002F (/), U+003D (=), and [noncharacters](https://infra.spec.whatwg.org/#noncharacter).

Is there a chance we could somehow escape the `alt` attribute?

Reply:
Hi, 
The text you're referring to in your quote is regarding the attribute **name** - an attribute name cannot contain the specified characters

> According to the [HTML standard](https://html.spec.whatwg.org/multipage/syntax.html#attributes-2), `>` is not a valid character in HTML attribute:
> 
> > Attributes have a name and a value. Attribute names must consist of one or more characters other than [controls](https://infra.spec.whatwg.org/#control), U+0020 SPACE, U+0022 ("), U+0027 ('), U+003E (>), U+002F (/), U+003D (=), and [noncharacters](https://infra.spec.whatwg.org/#noncharacter).

Attribute values on the other hand, are less restricted as long as the value is wrapped in matching quotes. There's a [specific text in the shared spec](https://html.spec.whatwg.org/multipage/introduction.html#restrictions-on-content-models-and-on-attribute-values:~:text=The%20attribute%20value%20can%20remain%20unquoted%20if%20it%20doesn%27t%20contain%20ASCII%20whitespace%20or%20any%20of%20%22%20%27%20%60%20%3D%20%3C%20or%20%3E.%20Otherwise%2C%20it%20has%20to%20be%20quoted%20using%20either%20single%20or%20double%20quotes.) clearly stating that `<` and `>` are supported as long as the attribute value is wrapped in quotes.

> The attribute value can remain [unquoted](https://html.spec.whatwg.org/multipage/syntax.html#unquoted) if it doesn't contain [ASCII whitespace](https://infra.spec.whatwg.org/#ascii-whitespace) or any of " ' ` = < or >. Otherwise, it has to be quoted using either single or double quotes.

In any case I'll look into it a bit further to see if there's anything I can do to mitigate the issue

Reply:
> The text you're referring to in your quote is regarding the attribute name 

@kidroca oops, you're right, sorry for mistake!

> In any case I'll look into it a bit further to see if there's anything I can do to mitigate the issue

I believe we could use a more complex HTML parser in Live Markdown.

cc @robertKozik who adapted ExpensiMark to Live Markdown.

Reply:
I've checked a bit. Why don't we just escape the values in attributes (as we are already escaping markdown syntax)? I don't have much context on your PR but it seems to work quite well, at least we are not getting an error, so we would have more time to work on better solution if needed.

Reply:
> Why don't we just escape the values in attributes (as we are already escaping markdown syntax)?

The parsing process applies rules sequentially. By the time the `image` rule executes, the original markdown has been transformed into HTML and incorporated into the attribute, resulting in elements like 
`<img src="..." alt="<code>...</code>" />`. Escaping attribute values initially would lead to 
`<img src="..." alt="&lt;code&gt;...&lt;/code&gt;" />`. This approach, however, complicates reverting HTML back to its original markdown, particularly when editing comments. Instead of restoring to the original markdown format ``[`...`](...)``, users would encounter a literal `[<code>...</code>](...)` in their edits.

---

I've identified an alternative solution that involves converting the HTML attribute back into the originally escaped markdown. You can find the implementation details here: https://github.com/Expensify/expensify-common/pull/676. It seems to work and should hopefully resolve any issues with the live markdown preview feature

Reply:
But does reverting back when editing work? When I've tested it yesterday I think typing ` ```test``` ` as img name, and trying to edit it leads to `test` string. I may be wrong tho, as I'm not remember exactly

Yeah your alternative solution will indeed solve our issues as we wouldn't have an `<` or `>` inside attributes. What is ETA for merging it?

Reply:
We seem to be in a bit of a dead lock situation here. To advance my `expensify-common` PR, I need to demonstrate its effectiveness within the `main` branch of NewDot. Would the solution I proposed here be acceptable to you: https://github.com/Expensify/App/issues/38952#issuecomment-2035196593?

Feel free to utilize this snapshot of `expensify-common` in your projects:
```json
"expensify-common": "git+ssh://git@github.com/kidroca/expensify-common.git#3ad1c3322f4ec8999483955afe6f55c95e78172a",
```
This snapshot reflects the changes in my PR for `expensify-common`.

1. Incorporating it into your PR would resolve your issue
2. It would allow QA to test the MD Image feature. 
  - But we should note to test this in the Tests section of your PR.
3. This in turn would allow approve the merge of the `expensify-common` PR
4. Letting you to finally switch back `expensify-common` to the Expensify's repo 

I'll be ready to cover any follow-up work related to the MD image feature.

What are your thoughts?

Reply:
It's a bit tricky because we're incorporating the `expensify-common` version inside `react-native-live-markdown` as well. So we would need to point to your branch within the new library version, which isn't my preferred approach. If possible, merging your `expensify-common` PR as it is would be great. If not, we can go with a patch containing your changes.

Here's how I envision the process:
1. Merge your `expensify-common` PR (if possible).
2. Publish a new version of `react-native-live-markdown` with both our and your fixes.
3. Create a joint PR for `react-native-live-markdown` and `expensify-common` in the E/App repo.

Reply:
@kidroca Apart from the one problem [I've mentioned here](https://github.com/Expensify/expensify-common/pull/676#issuecomment-2037523495) everything looks great. I think we can proceed with the patch inside the `react-native-live-markdown` so it will unblock you - does it sound good for you?